### ISSUE TITLE
OS X change wallpaper on all desktops

### ISSUE THREAD

**Issue by cjohnston1158** [bharadwaj-raju/WeatherDesk#34]
On OS X, running this will only change the wallpaper on the active desktop. Would be awesome to see it change the wallpaper on all desktops.

**Comment by bharadwaj-raju** [comment]
I thought this was done by:

```python
 OSX_SCRIPT = '''tell application "System Events"
                                  set desktopCount to count of desktops
                                    repeat with desktopNumber from 1 to desktopCount
                                      tell desktop desktopNumber
                                        set picture to POSIX file "%s"
                                      end tell
                                    end repeat
                                end tell
                              ''' % image
```

Which seems to do it to all desktops ... apparently not. Can you provide code for all desktops?

**Comment by cjohnston1158** [comment]
I can't, no. Maybe the original author of that change could?

**Comment by jnshnz** [comment]
Working on something similar here ([getspace.py](https://github.com/errorundefined/getspace)) – just found your repo, maybe we can cooperate on some things (will definitely be borrowing from your Linux functions after having had a comparable general idea).

With regards to this issue: that AppleScript bit is actually not really working as it used to. As [you can see](https://github.com/errorundefined/getspace/blob/master/getspace_osx.py). After reading [this](http://apple.stackexchange.com/a/145174) we are using a sqilte cmd at the moment (although that uses `killall Dock`, which is annoying if happening regularly).

Any other ideas/experiences?

**Comment by bharadwaj-raju** [comment]
Does `killall Dock` have a noticeable impact?

**Comment by jnshnz** [comment]
@bharadwaj-raju yes, it does, though it is somewhat negligible (some GUI effects will be disturbed/reset system wide for a brief instance, but no saved stuff will be deleted as far is I know).

I am researching the behavior of the `sqlite3 ~/Library/Application\ Support/Dock/desktoppicture.db "UPDATE data SET value='{}'" && killall Dock`-bit a little further, also because it is not reliable in a (changing) multi display environment.

Will let you know if I come up with something.

**Comment by bharadwaj-raju** [comment]
@jnshnz Then how does OS X do it through the GUI (which, I assume, causes no disturbance)?

Does `Dock` have a `--reload` or similar option? Or in AppleScript (imaginary: `tell application Dock to reload settings`)
